src/interpolation_methods.jl

```diff
@@ -1,7 +1,14 @@
 function _interpolate(A, t)
     ((t < A.t[1] || t > A.t[end]) && !A.extrapolate) &&
         throw(ExtrapolationError())
-    val, idx_prev = _interpolate(A, t, A.idx_prev[])
+    idx_guess = if hasfield(typeof(A), :seems_linear) && A.seems_linear
```

Comment:
It might be nicer to have all indexing logic in one place, that way you don't have to repeat it for integrals and derivatives. Say you leave this part unchanged, and refactor `get_idx` with this signature:

 `get_idx(A::AbstractInterpolation, t, iguess; lb = 1, ub_shift = -1, idx_shift = 0, side = :last)`

here you can get `tvec` and `seems_linear` from `A`, and overrule the given `iguess` with your one under the desired conditions.

Reply:
yeah, makes sense.

Reply:
Yes, and the bigger thing I haven't brought up is, it would then be great to make the OrdinaryDiffEq/DiffEqBase interpolations make use it of.

One step at a time 😅